Pipelines/templates/dotnet-publish-win-netcore-job.yml

```diff
@@ -0,0 +1,107 @@
+parameters:
+# Job Name
+- name: jobName
+  type: string
+  default: 'publish_win'
+# Optional Pre-Build Steps
+- name: 'preBuild'
+  type: stepList
+  default: []
+# Version of Dotnet SDK to use
+- name: dotnetVersion
+  type: string
+  default: '3.1.x'
+# Version of NuGet Tool to use
+- name: nugetVersion
+  type: string
+  default: '5.x'
+# Path to .csproj or .sln
+- name: projectPath
+  type: string
+  default: ''
+# Build Configuration
+- name: buildConfiguration
+  type: string
+  default: 'Release'
+# Project Name
+- name: projectName
+  type: string
+  default: ''
+# Release Version
+- name: releaseVersion
+  type: string
+  default: ''
+# Pipeline Artifact Name
+- name: artifactName
+  type: string
+  default: 'Unsigned_Binaries'
+
+jobs:
+- job: ${{ parameters.jobName }}
+  displayName: Dotnet Publish Win/NetCoreApp
+  pool:
+    vmImage: 'windows-latest'
+  steps:
+  - task: UseDotNet@2
+    displayName: Install Dotnet SDK
+    inputs:
+      packageType: 'sdk'
+      version: ${{ parameters.dotnetVersion }}
+  - task: NuGetToolInstaller@1
+    displayName: Install Nuget Tool
+    inputs:
+      versionSpec: ${{ parameters.nugetVersion }}
+  - bash: 'echo "##vso[task.setvariable variable=ReleaseVersion;]$RELEASE"'
+    env:
+      RELEASE: ${{ parameters.releaseVersion }}
+    displayName: Set Release Version
+  - ${{ parameters.preBuild }}
```

Comment:
The prebuild runs here to set the version but it looks like you set the version already on line 54.

It looks like we don't have to do any manual version trickery for the nuget ones.  I'm interested in why each works each way.

Reply:
Doing this to provide multiple ways of setting the release version. Not immediately useful, but may need it in the future.

Reply:
I think we only need one for now... I would choose the one that is actually being used and integrate that into the template.

Reply:
I'll remove the release version being set within the template then.

Reply:
Updated